app/forms/date_form.py

```diff
@@ -41,6 +48,13 @@ class DateForm(Form):
 
     validate_with += [DateCheck(date_messages['INVALID_DATE'])]
 
+    if 'minimum' in answer or 'maximum' in answer:
+        messages = None
+        if 'validation' in answer:
+            messages = answer['validation'].get('messages')
+        minimum_date, maximum_date = get_min_max_dates_for_single_date_period_validation(answer, answer_store, metadata)
+        validate_with += [SingleDatePeriodCheck(messages=messages, minimum_date=minimum_date, maximum_date=maximum_date)]
```

Comment:
Minor point - but let's not create a list just to append a single item to a list. Use `validate_with.append()`

Whilst there - maybe also change the same thing on line 49.